packages/task/src/browser/task-configurations.ts

```diff
@@ -129,16 +129,14 @@ export class TaskConfigurations implements Disposable {
      *
      * The invalid task configs are not returned.
      */
-    async getTasks(token: number): Promise<TaskConfiguration[]> {
+    async getTasks(): Promise<TaskConfiguration[]> {
         const configuredTasks = Array.from(this.tasksMap.values()).reduce((acc, labelConfigMap) => acc.concat(Array.from(labelConfigMap.values())), [] as TaskConfiguration[]);
         const detectedTasksAsConfigured: TaskConfiguration[] = [];
-        for (const [rootFolder, customizations] of Array.from(this.taskCustomizationMap.entries())) {
+        for (const [, customizations] of Array.from(this.taskCustomizationMap.entries())) {
             for (const cus of customizations) {
-                // TODO: getTasksToCustomize() will ask all task providers to contribute tasks. Doing this in a loop is bad.
-                const detected = await this.providedTaskConfigurations.getTaskToCustomize(token, cus, rootFolder);
+                const detected = this.taskDefinitionRegistry.getDefinition(cus);
```

Comment:
Now when I review my code again it seems that we should just add all the tasks from the `taskCustomizationMap` because any task added to `taskCustomizationMap` already passed the check of `isDetected`  inside `reorganizeTasks()` method.

**Am I missing something?** 
What was the purpose of the original (buggy?) filtering in `getTaskToCustomize()`